gallery/visualtests/nxMultiFileUpload.js

```diff
@@ -0,0 +1,545 @@
+        const [input] = await waitAndGetElements(`${complexExampleSelector} input[multiple]`);
+        await input.uploadFile(files.bytes);
+      });
+
+      const btnSelector = `${complexExampleSelector} .nx-selected-file__dismiss-btn`;
+
+      it('has a blue glow when focused', focusTest(complexExampleSelector, btnSelector));
+      it('has a dark border when hovered', hoverTest(complexExampleSelector, btnSelector));
+      it('has a dark border and blue glow when focused and hovered',
+          focusAndHoverTest(complexExampleSelector, btnSelector));
+      it('has a dark border and grey background when clicked', clickTest(complexExampleSelector, btnSelector));
+    });
+  });
+
+  describe('when required but empty and non-pristine', function() {
+    beforeEach(async function() {
+      const [input] = await waitAndGetElements(`${complexExampleSelector} input[multiple]`);
+      await input.uploadFile(files.bytes);
+
+      const [dismissBtn] = await waitAndGetElements(`${complexExampleSelector} .nx-selected-file__dismiss-btn`);
+      await dismissBtn.click();
+
+      expect(await isInDocument(dismissBtn)).toBe(false);
+    });
+
+    it('shows the validation error and the "No file selected" text turns red and gets and icon',
+        simpleTest(complexExampleSelector));
+  });
+
+  describe('functionality', function() {
+    async function openFileChooser(files) {
+      const [button] = await waitAndGetElements(`${complexExampleSelector} .nx-multi-file-upload__select-btn`);
+      const [fileChooser] = await Promise.all([getPage().waitForFileChooser(), button.click()]);
+      await fileChooser.accept(files);
+    }
+
+    it('opens file picker when the button is clicked, and displays all selected files', async function() {
+      await openFileChooser([files.bytes, files.kilobytes]);
+
+      const [selectedFile1] = (await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(1)`)),
+          textContent1 = await selectedFile1.evaluate(e => e.textContent);
+
+      expect(textContent1).toMatch(path.basename(files.bytes));
+      expect(textContent1).not.toMatch(path.dirname(files.bytes));
+      expect(textContent1).toMatch('14.0 B');
+
+      const [selectedFile2] = (await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(2)`)),
+          textContent2 = await selectedFile2.evaluate(e => e.textContent);
+
+      expect(textContent2).toMatch(path.basename(files.kilobytes));
+      expect(textContent2).not.toMatch(path.dirname(files.kilobytes));
+      expect(textContent2).toMatch('2.0 kB');
+    });
+
+    it('adds to selected files when file picker opened more than once', async function() {
+      await openFileChooser([files.bytes]);
+
+      const [selectedFile1] = (await waitAndGetElements(`${complexExampleSelector} .nx-selected-file:nth-child(1)`)),
+          textContent1 = await selectedFile1.evaluate(e => e.textContent);
+
+      expect(textContent1).toMatch('14.0 B');
```

Comment:
IMO these checks should be after the second file is added to the selection

Reply:
> IMO these checks should be after the second file is added to the selection

Fixed here: [1ef9e36](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/1ef9e36f7cf4a15b52a572334558e2cc9d3b0d89)

Reply:
> Fixed here: [1ef9e36](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/1ef9e36f7cf4a15b52a572334558e2cc9d3b0d89)

Doesn't seem like it is

Reply:
> > Fixed here: [1ef9e36](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/1ef9e36f7cf4a15b52a572334558e2cc9d3b0d89)
> 
> Doesn't seem like it is

Sorry about that, fixed here: [6a83eb6](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/6a83eb673acfd74d7287a9e364ba662c44efc154)

Reply:
> Sorry about that, fixed here: [6a83eb6](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/6a83eb673acfd74d7287a9e364ba662c44efc154)

You need to move the computation of the `textContent1` as well (and ideally the computation of `selectedFile1`.  Otherwise you're still testing the same thing as before, just later.